### ISSUE TITLE
Composer - "Split expense" and "Create expense" options only appear after a delay

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#54327]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.77-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5357293&group_by=cases:section_id&group_id=294998&group_order=asc
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team
**Device used:** Windows 10 / Chrome,  macOS Sequoia 15.0
**App Component:** Chat Report View

## Action Performed:
1. Admin: Navigate to https://staging.new.expensify.com/
2. Admin: Log in with a Gmail account that has a workspace
3. Admin: Navigate to Workspace settings - Members
4. Member: Log in with a new Gmail account on a different device
5. Admin: Invite the member to the workspace
6. Member: Quickly navigate to the workspace chat
7. Click on the composers "+" button
8. Dismiss the modal
9. After waiting a little, click on the composers "+" button again (on web, you might have to refresh the browser)

## Expected Result:
"Split expense" and "Create expense" options should be instantly available.

## Actual Result:
"Split expense" and "Create expense" options only appear after a delay for a new workspace member. On web, refreshing the browser restores the options quicker.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/d16fe11a-12bd-4e8a-8f19-5c2a9b6431f8

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869909199658687649</li>
        <li>Upwork Job ID: 1869909199658687649</li>
        <li>Last Price Increase: 2025-01-03</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @chiragsalian</details>

**Comment by vincdargento** [comment]
@trjExpensify FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by chiragsalian** [comment]
I'm unable to reproduce the issue. I see "Split expense" and "Create expense" just fine. I tested it as quickly as i could without refreshing the page.

Can you retest @vincdargento? I'm wondering if maybe there was some site slowdown causing you to miss the onyx data.

Also can you test @trjExpensify, if issue is non reproducible we can close it out.

**Comment by chiragsalian** [comment]
Asked tom and he is unable to reproduce the issue as well. Going to close it out, but if you are able to reproduce the issue let us know @vincdargento.

**Comment by trjExpensify** [comment]
Agreed. Thread [here](https://expensify.slack.com/archives/C07J32337/p1734563355627439) for the breadcrumbs!

**Comment by vincdargento** [comment]
@chiragsalian Tester was able to reproduce again, reopening in case you want to take a look

https://github.com/user-attachments/assets/99599ec7-9b41-4c23-9a69-f6e89a852252

**Comment by chiragsalian** [comment]
okay well either way since its not consistently reproducible by all, and/or since its an edge case i don't feel like its a blocker. So I'll remove out those labels. I'll mark it external to get more eyes on it.

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

"Split expense" and "Create expense" options only appear after a delay

### What is the root cause of that problem?

This is a backend issue: the reported data does not return the correct value and only updates to the correct value after refreshing or switching to another chat and then returning.

Here is the report data for this ticket case.

If we want to show `Split Expense`, the report must have a chat type of `policyExpenseChat` and `isOwnPolicyExpenseChat` must be `true`. However, in the report data, if `isOwnPolicyExpenseChat` is `false`, `Split Expense` will not be displayed.

https://github.com/Expensify/App/blob/f6c57593a0f5e6173835b0a6bde043d45455d733/src/libs/ReportUtils.ts#L7083

`Create Expense` is similar to `Split Expense` the condition to show it is not valid due to the report data, so Create Expense will not be displayed.”

https://github.com/Expensify/App/blob/f6c57593a0f5e6173835b0a6bde043d45455d733/src/libs/ReportUtils.ts#L7091

So, `moneyRequestOptionsList` will return an empty array, and this issue occurs.

https://github.com/Expensify/App/blob/c0959723544b62dff1ece709906674100276d3e2/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L177

### What changes do you think we should make in order to solve the problem?


To resolve this issue, we must check the `OpenReport` API to understand why it returns this value and ensure it responds with the correct value. If we want to fix it from the frontend, we need to add a condition: if the `chatType` is `policyExpenseChat` and `isOwnPolicyExpenseChat` is `false`, then we must refetch `openReport`.

https://github.com/Expensify/App/blob/129c98000cfe8f712dc8685c00d5c13c1125ced3/src/pages/home/ReportScreen.tsx#L537

```diff
    useEffect(() => {
        if (ReportUtils.isPolicyExpenseChat(report) && report?.isOwnPolicyExpenseChat) {
            return;
        }

        fetchReport();
    }, [fetchReportIfNeeded, report]);
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by trjExpensify** [comment]
@chiragsalian, do you agree with the RCA [here](https://github.com/Expensify/App/issues/54327#issuecomment-2563446228)? If so, let's remove the `Help wanted` label and switch `External` to `Internal`.

**Comment by trjExpensify** [comment]
Adding this to #quality as it's not reproducible 100% of the time.

**Comment by chiragsalian** [comment]
Yes i agree with the RCA and this is a backend issue. Updating labels.

**Comment by trjExpensify** [comment]
@chiragsalian what's the update here?

**Comment by chiragsalian** [comment]
No update just yet, i'm currently working on stability issues that are of higher priority than this.

**Comment by muttmuure** [comment]
Because this touches a CRITICAL flow and we're aiming generally to get interaction times down, I'm going to make it CRITICAL

**Comment by chiragsalian** [comment]
Honestly, im not sure. My plate is still pretty full.

**Comment by muttmuure** [comment]
This issue is currently CRITICAL, but it doesn’t feel like it since the feature generally works, just not as quickly as we’d like. Should we demote it?

**Comment by muttmuure** [comment]
Going to demote for now, if we feel passionately, let's up it to CRITICAL.

**Comment by trjExpensify** [comment]
I'm not super passionate, but if an invited member lands in the workspace chat, clicks the + icon they're directed to and doesn't see `Create expense` until after a refresh, that seems pretty bad in a mainline flow. That said, @huult is your RCA [here](https://github.com/Expensify/App/issues/54327#issuecomment-2563446228) alluding to the delay being only for `Split expense`?

**Comment by huult** [comment]
@trjExpensify This appears after we go back or refresh to we call `openReport` again to fetch new data, allowing `Split expense `to appear in this case. So, I suggested a solution: if `chatType` is `policyExpenseChat` and `isOwnPolicyExpenseChat` is `false`, we must re-call refetch to load up-to-date data from `openReport` or we need backend update `openReport`

**Comment by trjExpensify** [comment]
>@trjExpensify This appears after we go back or refresh to we call openReport again to fetch new data, allowing Split expense to appear in this case

Okay, so if it's only impacting `Split expense`, I'm cool with this being a Medium and not Critical.

**Comment by chiragsalian** [comment]
lower on my priority compared to other items so i haven't gotten to it yet.

**Comment by trjExpensify** [comment]
@chiragsalian what's the deal here?

**Comment by chiragsalian** [comment]
No update here yet. It feels lower on my priority list compared to others. Let me know if you think i should prioritize this differently. If its something that needs to be addressed sooner we may have to look for volunteers.

**Comment by trjExpensify** [comment]
Hard to say blind really, what are the other issues? This is a medium #quality issue in the pecking order for ref.

**Comment by chiragsalian** [comment]
For me, well i mostly cleared up my plate last week. So this week i want to tackle these,
1. [New Dot Onboarding for 1-50 Companies](https://github.com/Expensify/Expensify/issues/476894), (i'll begin detailed portion most likely tomorrow or day after) and
2. [Update ConciergeHealthChecker to better handle chats assigned to terminated agents](https://github.com/Expensify/Expensify/issues/445964) (i didnt set a proper k2 priority on this but it feels important to address soon and its been on my list for a while)

**Comment by trjExpensify** [comment]
Gotcha, gotcha. In the order of priority I'd say a medium #quality issue is ahead of a monthly issue not attached to any of the roadmap channels. 👍

**Comment by trjExpensify** [comment]
>Bumped to HIGH cuz it's affecting Create Expense

Is it? I thought we determined above it was `Split expense` only in the end at this point?

**Comment by mallenexpensify** [comment]
Thanks @trjExpensify , I wasn't certain and prefer to start with a _higher_ status then move down (vs the other way around where issues can easily get _lost_ as LOW).   

Updated status to LOW - Not very noticeable UX Reliability, API Reliability or Performance issues that only affect a single user or subset of users

**Comment by trjExpensify** [comment]
What's the latest here, @chiragsalian?

**Comment by chiragsalian** [comment]
between infra, bank audits and other priorities i havent been able to get to this. I've asked if someone else is able to take over here - https://expensify.slack.com/archives/C03TQ48KC/p1744742651964869

**Comment by marcaaron** [comment]
Gonna takeover for Chirag. I'll check this out today!

**Comment by marcaaron** [comment]
Ok, will have a fix up for this soon. I think it was introduced by this PR here: https://github.com/Expensify/Auth/pull/14555

**Comment by trjExpensify** [comment]
@marcaaron fixed this with an Auth PR I believe which is on prod, so I'm going to close it out.
